### PR TITLE
deposit: add identifiers fields

### LINKED ISSUE #766: Display `pids` and `identifiers` fields in the deposit form
Change the deposit form to display form inputs for `pids` and `identifiers` fields.

### PR DESCRIPTION
**Pull request by ntarocco**
* adds PIDs and identifiers fields to the deposit form
* closes #766

### CODE REVIEW

**Comment by ntarocco** [comment]
Blocked by release of: https://github.com/inveniosoftware/react-invenio-deposit/pull/262

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/RDMDepositForm.js
Comment (ppanero) [inline invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/RDMDepositForm.js:266]: ultra minor: `PIDS-FIXME` for easy grep

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/RDMDepositForm.js
Comment (ppanero) [inline invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/RDMDepositForm.js:279]: comment: This is a case of what I mentioned in my PR, this will not pass the validation phase because idutils does not have it.

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/RDMDepositForm.js
Comment (ppanero) [inline invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/RDMDepositForm.js:269]: Shall we only merge `DOI` since the other two were mainly a demo? Or we merge to be able to test, and remove before merging to master?

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/RDMDepositForm.js
Comment (ppanero) [inline invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/RDMDepositForm.js:515]: moderate: I think think related works keep going to the `related_identifiers` field (https://inveniordm.docs.cern.ch/reference/metadata/#related-identifiers-0-n).

**Review by ppanero — APPROVED** [review]
LGTM. Just clarify the doubt about *related works* before merging.

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/RDMDepositForm.js
Comment (ntarocco) [reply invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/RDMDepositForm.js:282]: Is this still the case?

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/RDMDepositForm.js
Comment (ntarocco) [reply invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/RDMDepositForm.js:269]: Yes I can do that, but in any case we need to come back and fix the injection because now it is hardcoded.
I will do that, waiting for the answers below to update the code.

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/RDMDepositForm.js
Comment (ppanero) [reply invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/RDMDepositForm.js:282]: A PR from @Pineirin will fix this in Marshmallow-Utils, and I will update Invenio-RDM-Records accordingly. So once that release chain goes in, this **will not** be a problem 🚀

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/RDMDepositForm.js
Comment (ppanero) [reply invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/RDMDepositForm.js:520]: If the user wants to change/customize it, needs to create a component. So in any case needs to touch code (there is no easy config way for him/her to do so). Then, I think is okay to do it as you did :)

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/RDMDepositForm.js
Comment (ntarocco) [reply invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/RDMDepositForm.js:269]: Done!

**Pull request merged**
